server/schedule/operator.go

```diff
@@ -521,21 +526,25 @@ func matchPeerSteps(cluster Cluster, source *core.RegionInfo, target *core.Regio
 		} else {
 			steps = append(steps, AddPeer{ToStore: id, PeerID: peer.Id})
 		}
+		if targetLeader == 0 {
+			targetLeader = id
+		}
 		kind |= OpRegion
 	}
 
 	// Check whether to transfer leader or not
 	intersection := getIntersectionStores(sourcePeers, targetPeers)
 	leaderID := source.GetLeader().GetStoreId()
-	isFound := false
 	for _, storeID := range intersection {
 		if storeID == leaderID {
-			isFound = true
+			targetLeader = 0
 			break
 		}
+		targetLeader = storeID
```

Comment:
It seems the `targetLeader` will be the last store in `intersection` instead of _first added follower_.

Reply:
yeah, if there is an intersection, no need to transfer to first added follower